include/arch/arm/arch.h

```diff
@@ -307,54 +307,58 @@ typedef u32_t k_mem_partition_attr_t;
 #ifdef CONFIG_ARM_USERSPACE
 #ifndef _ASMLANGUAGE
 /* Syscall invocation macros. arm-specific machine constraints used to ensure
- * args land in the proper registers. Currently, they are all stub functions
- * just for enabling CONFIG_USERSPACE on arm w/o errors.
+ * args land in the proper registers.
  */
+extern int _arm_do_syscall(u32_t call_id, u32_t arg1,
```

Comment:
Will this work?  `call_id` is the last argument passed by all the `_arch_syscall_invoke*()` functions, and the assembly implementation of this routine seems to take `r0` as `call_id` (I'm not familiar with the ABI).

Reply:
good catch! yes this isn't correct and I think will fail since the call_id is in the wrong position

Reply:
this prototype seems to be no longer necessary, remove